### PR TITLE
[GH-1215] Add snapshot_reference_id to Sarscov2IlluminaFull table schema

### PR DESCRIPTION
**Pull request by okotsopoulos**
### Purpose

- https://broadinstitute.atlassian.net/browse/GH-1215

After interacting with snapshots via the Rawls API, I realized that we would also want to store the snapshot's reference ID (unique identifier for snapshot reference within Terra Workspace).  Previously we only stored the snapshot ID (unique identifier for snapshot within TDR).

In testing this change, I ran into issues with my local Postgres.  Thanks @rexwangcc for the help!  As such, this PR is mostly documentation changes reflecting my experience, along with fixing some oddities I encountered in my general onboarding.

### Changes

Schema
- Added snapshot_reference_id to Sarscov2IlluminaFull table schema

Documentation
- Clarified order of operations when installing, using local Postgres
- Fixed broken intra-doc links
- Added docs readme with instruction for launching local documentation site

### Review Instructions

Schema
- In our mobbing 4/22, @rexwangcc said that because the existing changeset had not yet been applied in dev or prod environments, we could alter it to add a column rather than making a new changeset.
- **ASK:** Is it acceptable behavior to alter an existing changeset under these conditions?
- **ASK:** Will this cause issues for folks who have already applied Liquibase changes locally?
- **ASK:** I'd appreciate guidance for seeing this change locally: I am new to user-defined types.

Documentation
- If you wish, you can launch a local version of the documentation site (instructions in new readme).  I did this on my end and confirmed that I had fixed all intra-doc links, and that the formatting, flow and instructions in the development guideline page made sense.
- These changes look more involved than they are.  Changes related to Postgres are mainly reordering existing sections with slight alterations.

### CODE REVIEW

File: docs/md/index.md [hunk 1]
```diff
@@ -21,26 +21,27 @@ on Kubernetes clusters.
...
 
-The easiest way to build WFL is via `make` (or `gmake` on macOS), in addition, the
-following prerequisites are needed:
+The easiest way to build WFL is via `make`, in addition, the following
+prerequisites are needed:
 
 - [The Docker daemon](https://www.docker.com/products/docker-desktop)
 - Clojure (`brew install clojure` on macOS)
-- Python3 (`brew install python@3.8` on macOS)
+- Python3 (`brew install python@3.9` on macOS)
```
Comment (okotsopoulos) [inline docs/md/index.md:42]: Need to bump this: current version of Google Cloud SDK requires Python >= 3.9

File: docs/md/dev-process.md [hunk 2]
```diff
@@ -131,6 +139,59 @@ Here are some tips for WFL development.
...
+
+    ```
+    FATAL:  database "wfl" does not exist
+    ```
+
+    You should do as instructed within your terminal:
+
+    ```
+    createdb wfl
+    ```
```
Comment (rexwangcc) [inline docs/md/dev-process.md:182]: nit: might be helpful to also include `dropdb wfl` here in case folks want to recreate the local DBs?

**Review by rexwangcc — APPROVED** [review]
Thank you so much for the prompt fix on schema as well as the docs!!

re: questions:

> ASK: Is it acceptable behavior to alter an existing changeset under these conditions?

I personally think it's fine son long as: we haven't released this version AND we haven't propagate the changes to dev.

> ASK: Will this cause issues for folks who have already applied Liquibase changes locally?

I'm afraid so, folks might have to `dropdb wfl` and `createdb wfl` after this.

> ASK: I'd appreciate guidance for seeing this change locally: I am new to user-defined types.

Actually, this won't show up until you create a table from this `pipeline` type. I found a way to view the details from [StackOverflow](https://dba.stackexchange.com/questions/35497/display-user-defined-types-and-their-details):

```
WITH types AS (
    SELECT n.nspname,
            pg_catalog.format_type ( t.oid, NULL ) AS obj_name,
            CASE
                WHEN t.typrelid != 0 THEN CAST ( 'tuple' AS pg_catalog.text )
                WHEN t.typlen < 0 THEN CAST ( 'var' AS pg_catalog.text )
                ELSE CAST ( t.typlen AS pg_catalog.text )
                END AS obj_type,
            coalesce ( pg_catalog.obj_description ( t.oid, 'pg_type' ), '' ) AS description
        FROM pg_catalog.pg_type t
        JOIN pg_catalog.pg_namespace n
            ON n.oid = t.typnamespace
        WHERE ( t.typrelid = 0
                OR ( SELECT c.relkind = 'c'
                        FROM pg_catalog.pg_class c
                        WHERE c.oid = t.typrelid ) )
            AND NOT EXISTS (
                    SELECT 1
                        FROM pg_catalog.pg_type el
                        WHERE el.oid = t.typelem
                        AND el.typarray = t.oid )
            AND n.nspname <> 'pg_catalog'
            AND n.nspname <> 'information_schema'
            AND n.nspname !~ '^pg_toast'
),
cols AS (
    SELECT n.nspname::text AS schema_name,
            pg_catalog.format_type ( t.oid, NULL ) AS obj_name,
            a.attname::text AS column_name,
            pg_catalog.format_type ( a.atttypid, a.atttypmod ) AS data_type,
            a.attnotnull AS is_required,
            a.attnum AS ordinal_position,
            pg_catalog.col_description ( a.attrelid, a.attnum ) AS description
        FROM pg_catalog.pg_attribute a
        JOIN pg_catalog.pg_type t
            ON a.attrelid = t.typrelid
        JOIN pg_catalog.pg_namespace n
            ON ( n.oid = t.typnamespace )
        JOIN types
            ON ( types.nspname = n.nspname
                AND types.obj_name = pg_catalog.format_type ( t.oid, NULL ) )
        WHERE a.attnum > 0
            AND NOT a.attisdropped
)
SELECT cols.schema_name,
        cols.obj_name,
        cols.column_name,
        cols.data_type,
        cols.ordinal_position,
        cols.is_required,
        coalesce ( cols.description, '' ) AS description
    FROM cols
    ORDER BY cols.schema_name,
            cols.obj_name,
            cols.ordinal_position ;
```

from with in the Postgres Terminal (`psql wfl`). I'm sure there's a better way to do it…

File: docs/md/dev-process.md (hunk 2 above)
Comment (okotsopoulos) [reply docs/md/dev-process.md:182]: Added.

**Comment by okotsopoulos** [comment]
Verified new column present with a new wfl DB after applying Liquibase updates:

Screen Shot 2021-04-23 at 1 17 30 PM

**Review by tbl3rd — APPROVED** [review]
Thanks for the doc review recipe.

File: docs/readme.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+mkdocs serve
+```
+
+Logging output will indicate that changes are being detected,
+and point you to where documentation is served (e.g. http://127.0.0.1:8000).
+```
+INFO    -  Building documentation... 
+INFO    -  Cleaning site directory 
+WARNING -  A relative path to 'dev-sandbox.md' is included in the 'nav' configuration, which is not found in the documentation files 
+WARNING -  A relative path to 'dev-deployment.md' is included in the 'nav' configuration, which is not found in the documentation files
```
Comment (tbl3rd) [inline docs/readme.md:23]: Didn't notice this until I ran it myself.

**Review by tbl3rd — APPROVED** [review]

File: docs/readme.md (hunk 3 above)
Comment (okotsopoulos) [reply docs/readme.md:23]: I'm glad you mentioned this.  I thought that maybe I was missing some configuration on my end as a new person.  While I'm in here, I'll remove those undefined paths from the nav.

**Pull request merged**